stratum/hal/lib/tdi/tdi_sde_counter.cc

```diff
@@ -57,33 +57,28 @@ ::util::Status TdiSdeWrapper::WriteIndirectCounter(
   if (byte_count.has_value()) {
     tdi_id_t field_id;
     dataFieldInfo = table->tableInfoGet()->dataFieldGet(kCounterBytes);
-    RETURN_IF_NULL(dataFieldInfo);
-    field_id = dataFieldInfo->idGet();
-    RETURN_IF_TDI_ERROR(table_data->setValue(field_id, byte_count.value()));
+    if (dataFieldInfo) {
+      field_id = dataFieldInfo->idGet();
+      RETURN_IF_TDI_ERROR(table_data->setValue(field_id, byte_count.value()));
+    }
   }
   // Counter data: $COUNTER_SPEC_PKTS
   if (packet_count.has_value()) {
     tdi_id_t field_id;
     dataFieldInfo = table->tableInfoGet()->dataFieldGet(kCounterPackets);
-    RETURN_IF_NULL(dataFieldInfo);
-    field_id = dataFieldInfo->idGet();
-    RETURN_IF_TDI_ERROR(
-          table_data->setValue(field_id, packet_count.value()));
+    if (dataFieldInfo) {
+      field_id = dataFieldInfo->idGet();
+      RETURN_IF_TDI_ERROR(table_data->setValue(field_id, packet_count.value()));
+    }
   }
   const ::tdi::Device *device = nullptr;
   ::tdi::DevMgr::getInstance().deviceGet(dev_id, &device);
   std::unique_ptr<::tdi::Target> dev_tgt;
   device->createTarget(&dev_tgt);
 
   const auto flags = ::tdi::Flags(0);
-  if(byte_count.value() == 0 && packet_count.value() == 0) {
-    LOG(INFO) << "Resetting counters";
-    RETURN_IF_TDI_ERROR(table->clear(
-      *real_session->tdi_session_, *dev_tgt, flags));
-  } else {
-    RETURN_IF_TDI_ERROR(table->entryMod(
-     *real_session->tdi_session_, *dev_tgt, flags, *table_key, *table_data));
-  }
+  RETURN_IF_TDI_ERROR(table->entryMod(
```

Comment:
This is an API change. 
If I remember correctly, the reset counters needed a `table->clear()` call from SDE side. Has this changed?

Reply:
The method TdiSdeWrapper::WriteIndirectCounter is meant for resetting the single counter index. It is same as Drop 2 code. Even latest Stratum code also corrected the issue which was there with old code.  For your reference
https://github.com/stratum/stratum/blob/main/stratum/hal/lib/barefoot/bf_sde_wrapper.cc#L2678-L2720